packages/react-compose/src/compose.ts

```diff
@@ -5,7 +5,7 @@ import { wasComposedPreviously } from './wasComposedPreviously';
 import { mergeComposeOptions } from './mergeComposeOptions';
 
 function compose<
-  TElementType extends React.ElementType,
+  TElementType extends keyof JSX.IntrinsicElements,
```

Comment:
`TElementType` should be always an HTML tag as refs and props will be inferred from it.